app/controllers/school_groups_controller.rb

```diff
@@ -28,9 +28,17 @@ def comparisons
   end
 
   def priority_actions
-    service = SchoolGroups::PriorityActions.new(@school_group)
-    @priority_actions = service.priority_actions
-    @total_savings = sort_total_savings(service.total_savings)
+    respond_to do |format|
+      format.html do
+        service = SchoolGroups::PriorityActions.new(@school_group)
+        @priority_actions = service.priority_actions
+        @total_savings = sort_total_savings(service.total_savings)
+      end
+      format.csv do
+        send_data SchoolGroups::PriorityActionsCsvGenerator.new(school_group: @school_group).export,
+        filename: "#{@school_group.name} - #{I18n.t('school_groups.titles.priority_actions')} (#{Time.zone.now.strftime('%d/%m/%Y')}).csv"
```

Comment:
I'd be tempted to not use slashes in the time within the filename? and maybe not brackets or spaces either? (Or was this a requirement?)
`"#{@school_group.name}-#{I18n.t('school_groups.titles.priority_actions')}-#{Time.zone.now.strftime('%Y-%m-%d` ')}".parameterize + ".csv"

Reply:
:+1: thanks @urbanwide  will push an update going with your suggested change.